src/fu-util.c

```diff
@@ -1234,6 +1244,29 @@ fu_util_download_metadata (FuUtilPrivate *priv, GError **error)
 		if (!fu_util_download_metadata_enable_lvfs (priv, error))
 			return FALSE;
 	}
+
+	/* get devices from daemon */
+	devs = fwupd_client_get_devices (priv->client, NULL, error);
+	if (devs == NULL)
+		return FALSE;
+
+	/* get results */
+	for (guint i = 0; i < devs->len; i++) {
+		FwupdDevice *dev = g_ptr_array_index (devs, i);
+		if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_SUPPORTED))
+			devices_suppored_cnt++;
+	}
+
+	/* TRANSLATORS: success message -- where 'metadata' is information
+	 * about available firmware on the remote server */
+	g_string_append (str, _("Successfully downloaded new metadata: "));
+
+	/* TRANSLATORS: how many local devices can expect updates now */
+	g_string_append_printf (str, ngettext ("%u local device supported",
+					       "%u local devices supported",
+					       devices_suppored_cnt),
+					       devices_suppored_cnt);
```

Comment:
I'm confused, although I don't know much about `ngettext`, how does this select which string to show?

Reply:
It's a gettext thing -- it uses devices_suppored_cnt to work out the correct plural form.